### PR TITLE
lint: make errors more visible and consistent

### PR DESCRIPTION
**Pull request by benesch**
Make linter errors more visible by using the format

    lint: SEVERITY: program: message

where SEVERITY is either "error" or "fatal" and is printed in red.
Use this format consistently across all the lint and lint-like scripts,
like pycheck. Also make things quieter when successful, so the errors
are easier to spot.

In the course of thsi change, clean things up so that lint and pycheck
share more code, since they are very similar in nature.

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/4822)

### CODE REVIEW

File: bin/lint [hunk 1]
```diff
@@ -17,42 +17,23 @@ cd "$(dirname "$0")/.."
...
+        echo -e "hint: you can disable shellcheck locally by setting \$MZDEV_NO_SHELLCHECK=1" >&2
         exit 1
     fi
     version=$(shellcheck --version | grep version: | grep -oE "[0-9]\.[0-9]\.[0-9]" || echo "0.0.0+unknown")
     if ! version_compat "0.7.0" "$version"; then
-        printf "lint: fatal: shellcheck v0.7.0+ is required\n" >&2
-        printf "hint: detected version %q\n" "$version" >&2
-        printf "hint: you can disable shellcheck locally by setting \$MZDEV_NO_SHELLCHECK=1\n" >&2
+        echo -e "lint: $(red fatal:) shellcheck v0.7.0+ is required" >&2
+        echo -e "hint: detected version %q" "$version" >&2
```
Comment (quodlibetor) [inline bin/lint:30]: echo doesn't do quote (`%q`) substitution, IMO this should still be `printf` in case version is empty.

File: misc/lint/copyright.awk [hunk 2]
```diff
@@ -9,13 +9,18 @@
...
+
 function done()
 {
     if (!copyright) {
-        print "lint: copyright: " FILENAME " is missing copyright header" > "/dev/stderr"
+        err(FILENAME " is missing copyright header")
         exit 1
     } else if (copyright !~ /Copyright Materialize, Inc./) {
-        print "lint: copyright: " FILENAME " has malformatted copyright header" > "/dev/stderr"
+        err(FILENAME " has malformatted copyright header")
```
Comment (quodlibetor) [inline misc/lint/copyright.awk:23]: If this includes the `copyright` variable it will make it more obvious what's wrong, especially if the string `Copyright` appears in the first comment multiple times:

```suggestion
        err(FILENAME " has malformatted copyright header: " copyright)
```

**Review by quodlibetor — COMMENTED** [review]
This seems like it will make things _much_ easier to parse! 

Small comments, :shipit: when the one bug is fixed.

File: misc/lint/copyright.awk (hunk 2 above)
Comment (benesch) [reply misc/lint/copyright.awk:23]: I tried that, but it results in really long error messages like "lint: error: copyright: foo/bar/baz has malformatted copyright header: # Copyright SOME OTHER CO, Inc. All rights reserved" which I found hard to parse. Adjusted to print a hint mentioning the line number and the exact requirement.

File: bin/lint (hunk 1 above)
Comment (benesch) [reply bin/lint]: Oops, thanks, fixed.

**Comment by benesch** [comment]
TYVM for entertaining this @quodlibetor!
